src/OpenTelemetry.Extensions.Docker/Resources/DockerResourceDetector.cs

```diff
@@ -28,24 +28,33 @@ namespace OpenTelemetry.Extensions.Docker.Resources;
 public class DockerResourceDetector : IResourceDetector
 {
     private const string FILEPATH = "/proc/self/cgroup";
+    private const string FILEPATHV2 = "/proc/self/mountinfo";
+    private const string HOSTNAME = "hostname";
 
     /// <summary>
     /// Detects the resource attributes from Docker.
     /// </summary>
     /// <returns>Resource with key-value pairs of resource attributes.</returns>
     public Resource Detect()
     {
-        return this.BuildResource(FILEPATH);
+        var cGroupBuild = this.BuildResource(FILEPATH, true);
+        if (cGroupBuild == Resource.Empty)
+        {
+            cGroupBuild = this.BuildResource(FILEPATHV2, false);
+        }
+
+        return cGroupBuild;
     }
 
     /// <summary>
     /// Builds the resource attributes from Container Id in file path.
     /// </summary>
     /// <param name="path">File path where container id exists.</param>
+    /// <param name="isCGroupV1">Default parameter signifying where cgroup is of version 1 or 2. Defaults to version 1.</param>
     /// <returns>Returns Resource with list of key-value pairs of container resource attributes if container id exists else empty resource.</returns>
-    internal Resource BuildResource(string path)
+    internal Resource BuildResource(string path, bool isCGroupV1 = true)
```

Comment:
you already have two sets of methods: one for V1 and another for V2. I'd keep the same for this method too. That way you don't have to carry `isCGroupV1` parameter from method to method and you can call either BuildResourceV1() or BuildResourceV2() method in the Detect()